app/build.gradle

```diff
@@ -24,7 +24,7 @@ android {
         minSdkVersion 17
         targetSdkVersion 21
         versionCode 1
-        versionName "1.0"
+        versionName "1.0 - experimental"
```

Comment:
Seems to me that you need to get that information when building. Otherwise, this will print the experimental branch also when this is not experimental.
Needs to find a way how to get that from gradle and also add the git commit there.